pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -30,7 +30,7 @@
  */
 public class ApexOpenRedirectRule extends AbstractApexRule {
     private static final String PAGEREFERENCE = "PageReference";
-    private final Set<String> listOfStringLiteralVariables = new HashSet<>();
+    private static final Set<String> listOfStringLiteralVariables = new HashSet<>();
```

Comment:
don't use `static`. When running with multiple threads `-threads n` the partitioning is done over files, not rules, so n instances of this rule will coexist and share this store.